cmd/pulumi-test-language/l2resourcesimple_test.go

```diff
@@ -401,3 +401,48 @@ func TestL2SimpleResource_MissingRequiredPlugins(t *testing.T) {
 		"expected no error, got Error: unexpected required plugins: "+
 			"actual Set{language-mock@<nil>}, expected Set{language-mock@<nil>, resource-simple@2.0.0}")
 }
+
+// Run a simple successful test with a mocked runtime that edits the snapshot files.
+//
+// TODO(https://github.com/pulumi/pulumi/issues/13945): enable parallel tests
+//
+//nolint:paralleltest // These aren't yet safe to run in parallel
+func TestL2ResourceSnapshotEdit(t *testing.T) {
+	ctx := context.Background()
+	tempDir := t.TempDir()
+	engine := &languageTestServer{}
+	runtime := &L2ResourceSimpleLanguageHost{tempDir: tempDir}
+	handle, err := rpcutil.ServeWithOptions(rpcutil.ServeOptions{
+		Init: func(srv *grpc.Server) error {
+			pulumirpc.RegisterLanguageRuntimeServer(srv, runtime)
+			return nil
+		},
+	})
+	require.NoError(t, err)
+
+	prepareResponse, err := engine.PrepareLanguageTests(ctx, &testingrpc.PrepareLanguageTestsRequest{
+		LanguagePluginName:   "mock",
+		LanguagePluginTarget: fmt.Sprintf("127.0.0.1:%d", handle.Port),
+		TemporaryDirectory:   tempDir,
+		SnapshotDirectory:    "./testdata/snapshots_edit",
+		CoreSdkDirectory:     "sdk/dir",
+		SnapshotEdits: map[string]*testingrpc.PrepareLanguageTestsRequest_Replacement{
+			"test.txt": {
+				Pattern:     "testing",
```

Comment:
We never really end up testing this since we don't generate a `test.txt` with `testing` in it, or am I missing something here?

Reply:
We do in GeneratePackage

Reply:
Doh, I completely missed that.  This all checks out then, thanks for the pointer!